File: csharpcore-Verify.xunit/GildedRoseFinal/BackstageConcertPassItem.java
package gildedrose.item;

import gildedrose.Item;

public class BackstageConcertPassItem extends Item{
	private static final int CONCERT_CLOSE_DAYS = 10;
	private static final int CONCERT_VERY_CLOSE_DAYS = 5;
	private static final int QUALITY_INCREASE_VERY_CLOSE = 3;
	private static final int QUALITY_INCREASE_CLOSE = 2;
	private static final int QUALITY_INCREASE_NORMAL = 1;

	public BackstageConcertPassItem(int sellIn, int quality){
		super("Backstage passes to a TAFKAL80ETC concert", sellIn, quality);
	}
	
	@Override
	public void updateQuality(){
		if(isExpired()){
			setQuality(0);
		}else if(getSellInDays() < CONCERT_VERY_CLOSE_DAYS){
			increaseQuality(QUALITY_INCREASE_VERY_CLOSE);
		}else if(getSellInDays() < CONCERT_CLOSE_DAYS){
			increaseQuality(QUALITY_INCREASE_CLOSE);
		}else{
			increaseQuality(QUALITY_INCREASE_NORMAL);
		}
	}
}

File: csharpcore-Verify.xunit/GildedRoseFinal/BackstageConcertPassItem.cs
namespace GildedRoseKata;

/// <summary>
/// "Backstage passes", like aged brie, increases in Quality as its SellIn value approaches;
///
/// Quality increases by 2 when there are 10 days or less and by 3 when there are 5 days or less but Quality drops to 0 after the concert.
/// </summary>
public class BackstageConcertPassItem : Item
{
    private const int CONCERT_NEAR_DAYS = 10;
    private const int CONCERT_VERY_NEAR_DAYS = 5;
    private const int QUALITY_INCREASE_VERY_NEAR = 3;
    private const int QUALITY_INCREASE_NEAR = 2;
    private const int QUALITY_INCREASE_NORMAL = 1;

    public BackstageConcertPassItem(int sellIn, int quality) : base("Backstage passes to a TAFKAL80ETC concert", sellIn, quality)
    {
    }
    public override void UpdateQuality()
    {
        if (IsExpired())
        {
            Quality = 0;
            return;
        }

        switch (SellInDays)
        {
            case < CONCERT_VERY_NEAR_DAYS:
                IncreaseQuality(QUALITY_INCREASE_VERY_NEAR);
                break;

            case < CONCERT_NEAR_DAYS:
                IncreaseQuality(QUALITY_INCREASE_NEAR);
                break;

            default:
                // Default increase
                IncreaseQuality(QUALITY_INCREASE_NORMAL);
                break;
        }
    }
}